docling_sdg/qa/base.py

```diff
@@ -227,3 +235,41 @@ class GenQAC(QAPair[BaseModel]):
     critiques: dict[str, Critique] = Field(
         default={}, description="A set of critiques for each supported dimension."
     )
+
+def initialize_llm(llm_options: Optional[LlmOptions] = None) -> LLM:
+    if llm_options.api_key is None:
+        raise ValueError("API key is required")
+
+    match llm_options.provider:
+        case LlmProviders.OPENAI:
+            from llama_index.llms.openai import OpenAI
+
+            return OpenAI(
+                model=llm_options.model_id,
+                api_key=llm_options.api_key,
+                max_tokens=llm_options.max_new_tokens,
+                temperature=llm_options.additional_params[GenTextParamsMetaNames.TEMPERATURE],
+            )
+        case LlmProviders.OPENAI_LIKE:
+            from llama_index.llms.openai_like import OpenAILike
+
+            return OpenAILike(
+                model=llm_options.model_id,
+                api_base=llm_options.url,
+                api_key=llm_options.api_key,
+                max_tokens=llm_options.max_new_tokens,
+                temperature=llm_options.additional_params[GenTextParamsMetaNames.TEMPERATURE],
+            )
+        case LlmProviders.WATSONX:
+            if llm_options.project_id is None:
+                raise ValueError("Project ID is required")
+
+            from llama_index.llms.ibm import WatsonxLLM
+            return WatsonxLLM(
+                model_id=llm_options.model_id,
+                url=llm_options.url,
+                project_id=llm_options.project_id,
+                apikey=llm_options.api_key,
```

Comment:
pls change all the `api_key` assignment to:
`...=llm_options.api_key.get_secret_value()`, otherwise it gets the `*****` redacted string